### PR TITLE
Noise Reduction Script

### LINKED ISSUE #1102: Mini Audacity using python
**Is your feature request related to a problem? Please describe.**
Want to implement a feature that helps to filter an audio file by reducing the background noise similar to "Audacity".

**Describe the solution you'd like**
Using some of the python libraries to implement this "Mini Audacity" and the method used here is the spectral subtraction for speech processing.
* scipy.io.wavfile 
* numpy 
* matplotlib

### PR DESCRIPTION
**Pull request by A-kriti**
# Description

Added a new folder "Noise Reduction Script" containing all the required files.

## Type of change

- [✓] New feature (non-breaking change which adds functionality)

## Checklist:
- [✓] My code follows the style guidelines(Clean Code) of this project
- [✓] I have performed a self-review of my own code
- [✓] I have commented my code, particularly in hard-to-understand areas
- [✓] I have created a helpful and easy to understand README.md
- [✓] My documentation follows Template for README.md
- [✓] My changes generate no new warnings
- [✓] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

## Project Metadata

`` If there is no-file/nothing to fill the below fields with, then type: none ``

`` Example: `` If no requirements.txt needed/present, then type none in ``Requirments``.

Category:
- [ ] Calculators
- [ ] AI/ML
- [ ] Scrappers
- [ ] Social_Media
- [ ] PDF
- [ ] Image_Processing
- [ ] Video_Processing
- [ ] Games
- [ ] Networking
- [ ] OS_Utilities
- [ ] Automation
- [ ] Cryptography
- [ ] Computer_Vision
- [ ] Fun
- [x] Others

Title: Noise Reduction Script

Folder: Noise Reduction Script

Requirements: requirements.txt

Script: Noise Reduction Script.py

Arguments: none

Contributor: Akriti

Description: Removing background noise from an audio file just like "Audacity"

### CODE REVIEW

**Comment by santushtisharma10** [comment]
@A-kriti the script should be a python file. Also include requirements.txt file in the folder

**Comment by santushtisharma10** [comment]
And both the PR template is not correctly filled. Refer to approved PR and make changes accordingly

**Comment by A-kriti** [comment]
Done @santushtisharma10 ma'am!

**Comment by santushtisharma10** [comment]
requirements.txt is empty file. Refer to this [page](https://stackoverflow.com/questions/31684375/automatically-create-requirements-txt) to make a one.

**Comment by A-kriti** [comment]
@santushtisharma10 ma'am, it's empty in both the cases

Screenshot (407)

**Comment by santushtisharma10** [comment]
@A-kriti try using the command piqreqs [folder location] or you can manually write in the file.
image

the above commands worked for me
image

**Comment by A-kriti** [comment]
> @A-kriti try using the command piqreqs [folder location] or you can manually write in the file.
> image
> 
> the above commands worked for me
> image

@santushtisharma10 ma'am, thanks for this!

**Comment by santushtisharma10** [comment]
@A-kriti your code is almost same as this [code](https://github.com/shun60s/spectral-subtraction/blob/master/ss1.py). You just have added the code to display those graphs, rest the logic is exactly same as the this [code](https://github.com/shun60s/spectral-subtraction/blob/master/ss1.py)

**Comment by A-kriti** [comment]
@santushtisharma10 ma'am,  I had changed everything. Now, pls have a look!!

File: Mini Audacity/Mini Audacity.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Spectral Subtraction: Method used for noise reduction 
+
+import scipy.io.wavfile as wav
+import numpy as np
+import matplotlib.pyplot as plt
+
+file = input("Enter the path here :")
```
Comment (kaustubhgupta) [inline Mini Audacity/Mini Audacity.py:7]:
```suggestion
file = input("Enter the file path: ")
```

File: Mini Audacity/Mini Audacity.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+
+#finally plotting the graph showing the diffrence in the noise
+
+fig = plt.figure(figsize=(8,5))
+ax = plt.subplot(1,1,1)
+ax.plot(np.linspace(0,64000,64000),data,label='Original',color="orange")
+ax.plot(np.linspace(0,64000,64000),clean_data,label='Filtered',color="purple")
+ax.legend(fontsize=12)
+ax.set_title('Spectral Subtraction Method', fontsize=15)
```
Comment (kaustubhgupta) [inline Mini Audacity/Mini Audacity.py:61]: You have not displayed the graph to the user. Save the graph generated by the script in the same method as done in previous PR (audio file name (Spectral Subtraction graph).jpg) in the same folder `Mini Audacity`

File: Mini Audacity/Mini Audacity.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+
+cleaned_file = "Filtered_Audio.wav"   #final filtered audio
+wav.write(cleaned_file,rate=sr, data = clean_data.astype(np.int16))
```
Comment (kaustubhgupta) [inline Mini Audacity/Mini Audacity.py:65]: save the file in same folder `Mini Audacity` with name `Filtered Audio (original file name)`

File: Mini Audacity/Mini Audacity.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+
```
Comment (kaustubhgupta) [inline Mini Audacity/Mini Audacity.py:67]: remove 2 extra lines from the end

File: Mini Audacity/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+
+
```
Comment (kaustubhgupta) [inline Mini Audacity/README.md:2]: remove these blank lines

File: Mini Audacity/README.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+* Subtracting the noise spectral mean from input spectral, and istft (Inverse Short-Time Fourier Transform)
+* Finally getting an audio file with reduction in the background noise at a much higher extent
+
+## Output
+
+<img src="Graph/graph.jpg" height="500px">
+
+## Author(s)
+
+Akriti
```
Comment (kaustubhgupta) [inline Mini Audacity/README.md:33]:
```suggestion
[Akriti](https://github.com/A-kriti)
```

File: Mini Audacity/README.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+<img src="Graph/graph.jpg" height="500px">
+
+## Author(s)
+
+Akriti 
+
+
+
+
```
Comment (kaustubhgupta) [inline Mini Audacity/README.md:37]: remove these blank lines

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
- Change the folder and script name to 'Noise Reduction Script' as audacity has a lot of features and this script is only for noise reduction. Also, make the changes to the README file accordingly.

Other changes:

**Review by kaustubhgupta — COMMENTED** [review]
Also, update the PR template properly

**Review by kaustubhgupta — APPROVED** [review]
a bug is still there but anyways, due to less time, we will merge it.

**Review by santushtisharma10 — APPROVED** [review]

**Review by antrikshmisri — APPROVED** [review]

**Pull request merged**